chore(po): add update_po.py script to update PO/POT files

HastD opened the pull request:
The script at `files/po/update_po.py` carries out the following steps for each gettext domain (which currently only includes `audit_secureblue` but in the future will presumably include others):

* Use `xgettext` to update the POT file to reflect changes to the source code.
* Use `msginit` to update the English-language PO file (which doesn't require a translator).
* Use `msgmerge` to update other locales' PO files based on the updated POT file; new or modified strings will then need to be filled in by the translators.

The list of domains and their corresponding source files is stored in `files/po/po-source-files.json`.

These steps were previously done individually, and it's easier to just run this script, which also keeps track of which source files are associated to which gettext domain. PR authors will still need to run this script manually unless we figure out a way to automate it.

Also update PO/POT files using this script.

RoyalOughtness commented on files/po/update-po.sh:
```diff
@@ -0,0 +1,49 @@
+        | printf '%s\n%s\n' "$copyright_header" "$(cat -)" > "files/po/${domain}.pot"
+    msginit -i "files/po/${domain}.pot" -o "files/po/en/${domain}.po" --no-translator
+    for po_file in files/po/*/"$domain".po; do
+        if [[ $po_file != files/po/en/*.po ]]; then
```
does this expand properly? I would think this would always be false

HastD replied:
Yes, it does expand properly: https://www.gnu.org/software/bash/manual/bash.html#index-_005b_005b

> When the ‘`==`’ and ‘`!=`’ operators are used, the string to the right of the operator is considered a pattern and matched according to the rules described below in [Pattern Matching](https://www.gnu.org/software/bash/manual/bash.html#Pattern-Matching), as if the `extglob` shell option were enabled.

(Also, if it didn't expand properly, the script wouldn't have worked when I used it to update the PO files for this PR.)

HastD committed:
chore(po): add update_po.py script to update PO/POT files

The script at `files/po/update_po.py` carries out the following steps
for each gettext domain (which currently only includes
`audit_secureblue` but in the future will presumably include others):

* Use `xgettext` to update the POT file to reflect changes to the source
  code.
* Use `msginit` to update the English-language PO file (which doesn't
  require a translator).
* Use `msgmerge` to update other locales' PO files based on the updated
  POT file; new or modified strings will then need to be filled in by
  the translators.

The list of domains and their corresponding source files is stored in
`files/po/po-source-files.json`.

These steps were previously done individually, and it's easier to just
run this script, which also keeps track of which source files are
associated to which gettext domain. PR authors will still need to run
this script manually unless we figure out a way to automate it.

Also update PO/POT files using this script.

Signed-off-by: Daniel Hast <hast.daniel@protonmail.com>

RoyalOughtness committed:
Merge branch 'live' into po-update-script

HastD committed:
Merge branch 'live' into po-update-script
